src/Microsoft.OData.ModelBuilder/Vocabularies/VocabularyTermConfigurationShortcutsExtensions.cs

```diff
@@ -19,5 +21,149 @@ public static class VocabularyTermConfigurationShortcutsExtensions
         /// <returns></returns>
         public static PermissionTypeConfiguration HasScopes(this PermissionTypeConfiguration permissionTypeConfiguration, params string[] scopeNames)
             => permissionTypeConfiguration?.HasScopes(scopeNames.Select(scope => new ScopeTypeConfiguration().HasScope(scope)).ToArray());
+
+        /// <summary>
+        /// List of revisions
+        /// </summary>
+        /// <param name="navigationSourceConfiguration"></param>
+        /// <param name="description"></param>
+        /// <param name="version"></param>
+        /// <param name="date"></param>
+        /// <param name="removalDate"></param>
+        /// <returns></returns>
+        public static RevisionsConfiguration HasDeprecation<T>(
```

Comment:
>HasDeprecation [](http://example.com/codeflow?start=45&length=14)

Its name is "Revision", but function name is Deprecation?

Reply:
Deprecation is the revisionKind and because it has extra properties...Date and RemovalDate. So if you want to apply the deprecated revision kind to a model element then you use the HasDeprecation but for the other revisionKinds (Added and Modified) you use Has Revision

Reply:
solved